tests/SideBySide/AppConfig.cs

```diff
@@ -74,19 +73,8 @@ public static MySqlConnectionStringBuilder CreateSha256ConnectionStringBuilder()
 		// tests can run much slower in CI environments
 		public static int TimeoutDelayFactor { get; } = (Environment.GetEnvironmentVariable("APPVEYOR") == "True" || Environment.GetEnvironmentVariable("TRAVIS") == "true") ? 6 : 1;
 
-		private static string GetCodeRootPath()
-		{
-#if NET46
-			var currentAssembly = Assembly.GetExecutingAssembly();
-#else
-			var currentAssembly = typeof(AppConfig).GetTypeInfo().Assembly;
-#endif
-			var directory = new Uri(currentAssembly.CodeBase).LocalPath;
-			while (!string.Equals(Path.GetFileName(directory), "MySqlConnector", StringComparison.OrdinalIgnoreCase))
-				directory = Path.GetDirectoryName(directory);
-			return directory;
-		}
-
-		private static string ExpandVariables(string value) => value?.Replace("%TESTDATA%", TestDataPath);
+		private static string ExpandVariables(string value, string replacement) => value?.Replace("%TESTDATA%", replacement);
+		private static string ExpandLocalVariables(string value) => ExpandVariables(value, TestDataPath);
+		private static string ExpandRemoteVariables(string value) => ExpandVariables(value, RemoteTestDataPath);
```

Comment:
Secondly, I'm not sure there's a real win here for using `%TESTDATA%` as a variable when it's no longer automatically determined by the system, but is instead spelt out in the config file. Might as well just fully hard-code it in the config files in the values that need it.